packages/lesswrong/components/dialogues/DialoguesList.tsx

```diff
@@ -180,71 +276,60 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
+            <SettingsButton label={``} onClick={() => setShowSettings(!showSettings)}/>
+          </LWTooltip>
+        </div>
       )}
+      
       </SectionTitle>
+      {showSettings && currentUser && <DialoguesSectionFrontpageSettings
+          persistentSettings={true}
+          hidden={false}
+          currentShowDialogues={currentUser.showDialoguesList}
+          currentShowMyDialogues={currentUser.showMyDialogues}
+          currentShowMatches={currentUser.showMatches}
+          currentShowRecommendedPartners={currentUser.showRecommendedPartners}
+          hideReciprocityButtons={!showReciprocityRecommendations}
+        />}
 
       {dialogueMatchmakingEnabled.get() && <AnalyticsContext pageSubSectionContext="frontpageDialogueMatchmaking">
-        <div>
-          {currentUser && rowPropsList?.map(rowProps => {
-            const { targetUser, checkId, userIsChecked, userIsMatched } = rowProps;
-            return (<div key={targetUser._id} className={classes.dialogueUserRow}>
-              <div className={classes.dialogueLeftContainer}>
-                <div className={classes.dialogueMatchCheckbox}>
-                  <DialogueCheckBox
-                    targetUserId={targetUser._id}
-                    targetUserDisplayName={targetUser.displayName}
-                    checkId={checkId}
-                    isChecked={userIsChecked}
-                    isMatched={userIsMatched}
-                  />
-                </div>
-                <PostsItem2MetaInfo className={classes.dialogueMatchUsername}>
-                  <UsersName
-                    documentId={targetUser._id}
-                    simple={false}
-                  />
-                </PostsItem2MetaInfo>
-                <PostsItem2MetaInfo className={classes.dialogueMatchNote}>
-                  You've matched with this user
-                </PostsItem2MetaInfo>
-              </div>
-              <div className={classes.dialogueRightContainer}>
-                <div className={classes.dialogueMatchMessageButton}>
-                  <MessageButton
-                    targetUserId={targetUser._id}
-                    currentUser={currentUser}
-                    isMatched={userIsMatched}
-                  />
-                </div>
-                <div className={classes.dialogueMatchPreferencesButton}>
-                  <DialogueNextStepsButton
-                    isMatched={userIsMatched}
-                    checkId={checkId}
-                    targetUserId={targetUser._id}
-                    targetUserDisplayName={targetUser.displayName}
-                    currentUser={currentUser}
-                  />
-                </div>
-              </div>
-            </div>);
-          })}
-        </div>
+        {showReciprocityRecommendations && <div>
+          { (currentUser?.showMatches || currentUser?.showRecommendedPartners ) &&
+            <div className={classes.explanatoryNoteBox}>
```

Comment:
This is nice, but kind of ends up looking like it applies to the whole Dialogue section?

Reply:
<img width="844" alt="image" src="https://github.com/ForumMagnum/ForumMagnum/assets/12041390/ded5af49-284b-4eea-8a1c-4572768b1dbb">